svf/lib/WPA/Andersen.cpp

```diff
@@ -709,24 +709,20 @@ inline void Andersen::collapseFields()
  */
 void Andersen::mergeSccCycle()
 {
-    NodeStack revTopoOrder;
-    NodeStack & topoOrder = getSCCDetector()->topoNodeStack();
-    while (!topoOrder.empty())
+    NodeStack revTopoOrder = getSCCDetector()->revTopoNodeStack();
+    /// referenced topoOrder is used for restoring the topological order for later solving.
+    NodeStack &topoOrder = getSCCDetector()->topoNodeStack();
```

Comment:
`topoOrder` has been there when you retrieve it and should not be changed later at line 725.